### PR TITLE
Proj LWD Rebuild Respects Deleted Rows

### PR DESCRIPTION
**Pull request by jsc0218**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
fix the bug that didn't consider _row_exists column in rebuild option of projection lightweight delete

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 0feb4651d8cb0e31c2a3785dee8cf209e3dd127a with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71089/0feb4651d8cb0e31c2a3785dee8cf209e3dd127a/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/c82cf25b3e5864bcc153cbe45adb8c6527e1ec6e/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Storages/ProjectionsDescription.cpp [hunk 1]
```diff
@@ -294,8 +294,23 @@ Block ProjectionDescription::calculate(const Block & block, ContextPtr context)
...
+    {
+        query_ast_copy = query_ast->clone();
+        auto * select_row_exists = query_ast_copy->as<ASTSelectQuery>();
+        if (!select_row_exists)
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot get ASTSelectQuery when adding _row_exists = 1. It's a bug");
+
+        select_row_exists->setExpression(
+            ASTSelectQuery::Expression::WHERE,
+            makeASTFunction("equals", std::make_shared<ASTIdentifier>("_row_exists"), std::make_shared<ASTLiteral>(1)));
+        // std::cout<<serializeAST(*query_ast_copy)<<std::endl;
```
Comment (CurtizJ) [inline src/Storages/ProjectionsDescription.cpp:309]: Please remove

**Review by CurtizJ — APPROVED** [review]

**Comment by jsc0218** [comment]
CI [54748](https://github.com/ClickHouse/ClickHouse/issues/54748)

**Comment by clickgapai** [comment]
Hi @jsc0218 @CurtizJ — while reviewing this PR I found the following:

- **Bug (P1):** Normal projections (ORDER BY) ignore `_row_exists` filter during LWD rebuild because `FetchColumns` stage skips WHERE execution — https://github.com/ClickHouse/ClickHouse/issues/104664

Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**